sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -0,0 +1,391 @@
+    coef = [1., 2.]
+    X = np.array([[1, 1, 1, 1, 1], [0, 1, 2, 3, 4]]).T
+    y = np.dot(X, coef)
+    glm = GeneralizedLinearRegressor(alpha=0, family='normal', link='identity',
+                                     fit_intercept=False, solver=solver,
+                                     tol=1e-7)
+    res = glm.fit(X, y)
+    assert_allclose(res.coef_, coef, rtol=1e-6)
+
+
+@pytest.mark.parametrize(
+    'family',
+    [NormalDistribution(), PoissonDistribution(),
+     GammaDistribution(), InverseGaussianDistribution(),
+     TweedieDistribution(power=1.5), TweedieDistribution(power=4.5)])
+@pytest.mark.parametrize('solver, tol', [('lbfgs', 1e-6)])
+def test_glm_log_regression(family, solver, tol):
+    """Test GLM regression with log link on a simple dataset."""
+    coef = [0.2, -0.1]
+    X = np.array([[1, 1, 1, 1, 1], [0, 1, 2, 3, 4]]).T
+    y = np.exp(np.dot(X, coef))
+    glm = GeneralizedLinearRegressor(
+                alpha=0, family=family, link='log', fit_intercept=False,
+                solver=solver, tol=tol)
+    res = glm.fit(X, y)
+    assert_allclose(res.coef_, coef, rtol=5e-6)
+
+
+@pytest.mark.parametrize('fit_intercept', [True, False])
+def test_warm_start(fit_intercept):
+    n_samples, n_features = 100, 10
+    n_predict = 10
+    X, y, coef = make_regression(n_samples=n_samples+n_predict,
+                                 n_features=n_features,
+                                 n_informative=n_features-2, noise=0.5,
+                                 coef=True, random_state=42)
+
+    glm1 = GeneralizedLinearRegressor(
+        warm_start=False,
+        fit_intercept=fit_intercept,
+        max_iter=1000
+    )
+    glm1.fit(X, y)
+
+    glm2 = GeneralizedLinearRegressor(
+        warm_start=True,
+        fit_intercept=fit_intercept,
+        max_iter=1
+    )
+    glm2.fit(X, y)
+    assert glm1.score(X, y) > glm2.score(X, y)
+    glm2.set_params(max_iter=1000)
+    glm2.fit(X, y)
+    assert_allclose(glm1.coef_, glm2.coef_, rtol=1e-4, atol=1e-5)
+    assert_allclose(glm1.score(X, y), glm2.score(X, y), rtol=1e-4)
+    # TODO: investigate why this doesn't match
+    # assert glm1.n_iter_ == glm2.n_iter_ + 2
+
+
+@pytest.mark.parametrize('n_samples, n_features', [(100, 10), (10, 100)])
+@pytest.mark.parametrize('fit_intercept', [True, False])
+@pytest.mark.parametrize('solver', GLM_SOLVERS)
+def test_normal_ridge_comparison(n_samples, n_features, fit_intercept, solver):
+    """Test ridge regression for Normal distributions.
+
+    Case n_samples >> n_features
+
+    Compare to test_ridge in test_ridge.py.
+    """
+    alpha = 1.0
+    n_predict = 10
+    X, y, coef = make_regression(n_samples=n_samples+n_predict,
+                                 n_features=n_features,
+                                 n_informative=n_features-2, noise=0.5,
+                                 coef=True, random_state=42)
+    y = y[0:n_samples]
+    X, T = X[0:n_samples], X[n_samples:]
```

Comment:
Use `train_test_split` and get rid of `n_predict`.

Also let's call them `X_train, X_test, y_train, y_test`